guide/src/ref/packet-log.md

```diff
@@ -1,37 +1,167 @@
+
+### Convert a Rattan packet log
+
+#### Rust subcommand
+
+To convert a Rattan packet log to `.pcapng` file, which can be viewed with Wireshark or other tools, we provide
+a subcommand `convert`.
 
-We also provide a python script [scripts/log_converter.py](https://github.com/stack-rs/rattan/blob/main/scripts/log_converter.py) to convert the packet log file (along with the flow description file) to some pcapng files, which can be viewed with Wireshark or other tools. You can run the script like this:
+Example usage:
 
 ```bash
-python3 scripts/log_converter.py packet.log converted.pcapng
+rattan convert ./packet.rtl [./output]
 ```
 
-This will read `packet.log` and `packet.log.flows`, and generate a pcapng file for each end of Rattan (identified by IP address), with the name `converted_<ip>.pcapng` (e.g. `converted_10_1_1_1.pcapng`).
+This expects the following files exist, as input:
 
-Note that some fields are artificially constructed in the pcapng file, as the compressed packet log only stores partial information. Artificial fields include:
+- `./packet.rtl`
+- `./packet.flows`
+- `./packet.raw` (optional, only if the packet log was recorded in `raw-tcp` or `raw-ip` mode).
+
+And it will Generated the following files, as output, if the address of the two ends of the emulated link
+of Rattan was `10.1.1.1` and `10.2.1.1`":
+
+- `./output_10_1_1_1.pcapng`
+- `./output_10_2_1_1.pcapng`
+
+If the second argumnet was not provided(the output prefix), the default value for it is the first argument (the path
+to the `.rtl` file).
+
+Note that, in `compact-tcp` mode, some fields are artificially constructed in the pcapng file, as the compressed
+packet log only stores partial information. Artificial fields include:
 
 - `Window`: The TCP window size is set to `65535` for all packets.
 - `Checksum`: The TCP checksum is set to `0` for all packets.
 - `Urgent Pointer`: The TCP urgent pointer is set to `0` for all packets.
 - `Options`: The TCP header length is real, but option content is set to `0` for all packets.
 
+Besides, in `raw-tcp` mode, some fields are artifically constructed in the pcapng file, since per-packet
+info in IP headers are not recorded at all. Artificial fields include:
+
+- `Identification`: Set to `0`.
+- `Flags` : Set to `Don't Fragment`.
+- `Checksum` : Set to `0`, or disabled.
+
+#### Python script
+
+We provide a python script (scripts/log_converter.py), that provides basically the same function as the above-mentioned Rust subcommand does.
+Notice that, the Python script can be slower than the Rust subcommand for one to
+two orders of magnitude when processing large log files.
+
+Example usage:
+
+```bash
+python scripts/log_converter.py ./packet.rtl ./output
+```
+
+The paths to input and output files are the same as those for `rattan convert`.
+
+The behaviour of the python script is basically the same as `rattan convert`,
+except for that the output path prefix is not optional.
```

Comment:
We might also make it optional I think, so to make the interfaces of our converter consistent.